### PR TITLE
Add support for decoding tiff images with CieLab color space

### PR DESCRIPTION
**Pull request by brianpopow**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

This PR adds support for decoding Tiff images with the CieLab color space.

### CODE REVIEW

File: src/ImageSharp/Formats/Tiff/PhotometricInterpretation/CieLabPlanarTiffColor{TPixel}.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace SixLabors.ImageSharp.Formats.Tiff.PhotometricInterpretation
+{
+    /// <summary>
+    /// Implements decoding pixel data with photometric interpretation of type 'CieLab' with the planar configuration.
+    /// </summary>
+    internal class CieLabPlanarTiffColor<TPixel> : TiffBasePlanarColorDecoder<TPixel>
+        where TPixel : unmanaged, IPixel<TPixel>
+    {
+        private readonly ColorSpaceConverter colorSpaceConverter = new();
```
Comment (gfoidl) [inline src/ImageSharp/Formats/Tiff/PhotometricInterpretation/CieLabPlanarTiffColor{TPixel}.cs:20]:
```suggestion
        private static readonly ColorSpaceConverter colorSpaceConverter = new();
```
?

File: src/ImageSharp/Formats/Tiff/PhotometricInterpretation/CieLabTiffColor{TPixel}.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace SixLabors.ImageSharp.Formats.Tiff.PhotometricInterpretation
+{
+    /// <summary>
+    /// Implements decoding pixel data with photometric interpretation of type 'CieLab'.
+    /// </summary>
+    internal class CieLabTiffColor<TPixel> : TiffBaseColorDecoder<TPixel>
+        where TPixel : unmanaged, IPixel<TPixel>
+    {
+        private readonly ColorSpaceConverter colorSpaceConverter = new();
```
Comment (gfoidl) [inline src/ImageSharp/Formats/Tiff/PhotometricInterpretation/CieLabTiffColor{TPixel}.cs:19]:
```suggestion
        private static readonly ColorSpaceConverter colorSpaceConverter = new();
```

File: tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs [hunk 3]
```diff
@@ -318,6 +318,20 @@ public void TiffDecoder_CanDecode_YCbCr_24Bit<TPixel>(TestImageProvider<TPixel>
...
 
+        [Theory]
+        [WithFile(CieLab, PixelTypes.Rgba32)]
+        [WithFile(CieLabPlanar, PixelTypes.Rgba32)]
+        [WithFile(CieLabLzwPredictor, PixelTypes.Rgba32)]
+        public void TiffDecoder_CanDecode_CieLab<TPixel>(TestImageProvider<TPixel> provider)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            // Note: The image from MagickReferenceDecoder does not look right, maybe we are doing something wrong
+            // converting the pixel data from Magick.NET to our format with CieLab?
```
Comment (JimBobSquarePants) [inline tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs:329]: What's happening here?

**Review by JimBobSquarePants — APPROVED** [review]
Just checked the ref vs input. Looks great! :+1:  Nice work!

File: tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs (hunk 3 above)
Comment (brianpopow) [reply tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs:329]: The image `Tiff\CieLab.tiff` looks wrong when using the `MagickReferenceDecoder`:
output

It should look like this:
expected

ImageMagick itself can decode this image just fine, that's why I thought we maybe doing something wrong in `MagickReferenceDecoder`.

Comment (JimBobSquarePants) [reply tests/ImageSharp.Tests/Formats/Tiff/TiffDecoderTests.cs:329]: Ah right.... Yep, 

`IMagickImage<T>` contains `ColorSpace` enum which we are blindly ignoring when converting. I always assumed they did the same as us and converted color spaces on decode.

It looks like we should be checking that first and converting on the fly. 

@dlemstra Am I correct here?

**Pull request merged**